client/finality-grandpa/src/authorities.rs

```diff
@@ -631,6 +686,36 @@ impl<H, N: Add<Output=N> + Clone> PendingChange<H, N> {
 	}
 }
 
+// Tracks authority set changes. We store the block numbers for the first block of each authority
+// set.
+#[derive(Debug, Encode, Decode, Clone, PartialEq)]
+pub struct AuthoritySetChanges<N> {
+    authority_set_changes: Vec<(u64, N)>, // (set_id, block_number)
```

Comment:
I'm sort of thinking about storing this as `(set_id, (first_block, last_block))` to make the consistency checks easier for when the set of changes is incomplete

Reply:
I guess we could just check that this vec is missing entries for older set ids to figure out that we don't have all data. E.g. if we have set id 3 in there then we should have all the ones before that as well contiguosly.